packages/components/src/ui/context/context-connect.ts

```diff
@@ -66,18 +94,12 @@ export function contextConnect< P >(
 		mergedNamespace = [ ...mergedNamespace, namespace ];
 	}
 
-	WrappedComponent.displayName = namespace;
-
-	// @ts-ignore internal property
-	WrappedComponent[ CONNECT_STATIC_NAMESPACE ] = [
-		...new Set( mergedNamespace ),
-	];
-
-	// @ts-ignore WordPressComponent property
-	WrappedComponent.selector = `.${ getStyledClassNameFromKey( namespace ) }`;
-
-	// @ts-ignore
-	return WrappedComponent;
+	// @ts-expect-error
+	return Object.assign( WrappedComponent, {
+		[ CONNECT_STATIC_NAMESPACE ]: [ ...new Set( mergedNamespace ) ],
+		displayName: namespace,
+		selector: `.${ getStyledClassNameFromKey( namespace ) }`,
+	} );
```

Comment:
Just merged together for readability.

Aside: I also looked into whether we could somehow depend on inference for this return type, rather than forcing a `WordPressComponentProps` type. Apparently the blocker is that our components need to support the `as` prop, which will dynamically change the set of accepted props. All the `as`-related code errors if we rely on inference here.

Reply:
Interesting. Maybe we should add this finding as an inline comment?